client/settings/account-details/index.js

```diff
@@ -3,50 +3,69 @@ import { createInterpolateElement } from '@wordpress/element';
 import React from 'react';
 import SectionStatus from '../../components/section-status';
 import './style.scss';
+import { useAccount, useGetCapabilities } from 'wcstripe/data/account';
+
+const useIsCardPaymentsEnabled = () =>
+	useGetCapabilities().card_payments === 'active';
+
+const useAreDepositsEnabled = () => {
+	const { data } = useAccount();
+
+	return (
+		data.account?.payouts_enabled &&
+		Boolean( data.account?.settings?.payouts?.schedule?.interval )
+	);
+};
+
+const PaymentsSection = () => {
+	const isEnabled = useIsCardPaymentsEnabled();
 
-const PaymentsSection = ( props ) => {
 	return (
 		<div className="account-details__row">
 			<p>{ __( 'Payments:', 'woocommerce-gateway-stripe' ) }</p>
-			<SectionStatus { ...props } />
+			<SectionStatus isEnabled={ isEnabled } />
 		</div>
 	);
 };
 
-const DepositsSection = ( props ) => {
+const DepositsSection = () => {
+	const isEnabled = useAreDepositsEnabled();
+
 	return (
 		<div className="account-details__row">
 			<p>{ __( 'Deposits:', 'woocommerce-gateway-stripe' ) }</p>
-			<SectionStatus { ...props } />
+			<SectionStatus isEnabled={ isEnabled } />
 		</div>
 	);
 };
 
-const MissingAccountDetailsDescription = ( { accountStatus } ) => {
-	const { accountLink, paymentsEnabled, depositsEnabled } = accountStatus;
+const MissingAccountDetailsDescription = () => {
+	const isPaymentsEnabled = useIsCardPaymentsEnabled();
+	const areDepositsEnabled = useAreDepositsEnabled();
 
-	let description = '';
-	if ( ! paymentsEnabled || ! depositsEnabled ) {
-		description = createInterpolateElement(
-			/* translators: <a> - dashboard login URL */
-			__(
-				'Payments/deposits may be disabled for this account until missing business information is updated. <a>Update now</a>',
-				'woocommerce-gateway-stripe'
-			),
-			// eslint-disable-next-line jsx-a11y/anchor-has-content
-			{ a: <a href={ accountLink } /> }
-		);
-	}
-
-	if ( ! description ) {
+	if ( isPaymentsEnabled && areDepositsEnabled ) {
 		return null;
 	}
 
-	return <div className="account-details__desc">{ description }</div>;
+	return (
+		<div className="account-details__desc">
+			{ createInterpolateElement(
+				/* translators: <a> - dashboard login URL */
+				__(
+					'Payments/deposits may be disabled for this account until missing business information is updated. <a>Update now</a>',
+					'woocommerce-gateway-stripe'
+				),
+				// eslint-disable-next-line jsx-a11y/anchor-has-content
+				{ a: <a href="https://stripe.com/support" /> }
+			) }
+		</div>
+	);
 };
 
-const AccountDetails = ( { accountStatus } ) => {
-	if ( accountStatus.error ) {
+const AccountDetails = () => {
+	const { data } = useAccount();
+
+	if ( Object.keys( data.account ?? {} ).length === 0 ) {
```

Comment:
I had to compile it in my head before understanding and I'd rather not have to :smile:. Could we make this more obvious somehow? Or maybe extract to a variable, like `isAccountLoaded`?

Reply:
That's a good idea - renamed to `hasAccountError` (since the error message is "Error determining the account connection status."). Here we're checking whether we have errors with the account data.